### PR TITLE
Refactor spinwick creation logic

### PR DESCRIPTION
**Pull request by gabrieljackson**
This change cleans up core spinwick creation logic in preparation for adding tests. The only user experience change should be that more error details are surfaced.

Fixes https://mattermost.atlassian.net/browse/CLD-6732

```release-note
Refactor spinwick creation logic
```

### CODE REVIEW

File: internal/cloudtools/installation.go [hunk 1]
```diff
@@ -1,39 +1,34 @@
 package cloudtools
 
 import (
-	"fmt"
-
-	cloudModel "github.com/mattermost/mattermost-cloud/model"
+	cloud "github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
```
Comment (fmartingr) [inline internal/cloudtools/installation.go:5]: Archived for three years now. At some point we will need to start migrating codebases to `fmt.Errorf("%w", err)`

File: server/builds.go [hunk 2]
```diff
@@ -39,23 +39,22 @@ func (b *Builds) dockerRegistryClient(s *Server) (reg *registry.Registry, err er
...
-func (b *Builds) waitForImage(ctx context.Context, s *Server, reg *registry.Registry, pr *model.PullRequest, imageToCheck string, logger logrus.FieldLogger) (*model.PullRequest, error) {
-	desiredTag := b.getInstallationVersion(pr)
+func (b *Builds) waitForImage(ctx context.Context, reg *registry.Registry, desiredTag, imageToCheck string, logger logrus.FieldLogger) error {
 	logger = logger.WithFields(logrus.Fields{"image": imageToCheck, "tag": desiredTag})
 
 	for {
 		select {
 		case <-ctx.Done():
-			return pr, errors.New("timed out waiting for image to publish")
+			return errors.New("timed out waiting for image to publish")
```
Comment (fmartingr) [inline server/builds.go:48]: Should `waitForImage` handle its own timeout? Like... let's allow 20m for a image to be created but a global timeout of 45m for the entire process on the parent context. (?)

Just exposing what comes to mind. No need to make any changes.

**Review by fmartingr — APPROVED** [review]
LGTM! I just left a few comments that do not need to be addressed for this to be merged, logic seems good and changes make sense 👍

File: server/builds.go (hunk 2 above)
Comment (gabrieljackson) [reply server/builds.go:48]: Yeah, that probably makes sense to do here. I will create a ticket so we remember to add it in our next improvement. Thanks!

**Pull request merged**